matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerExporter.java

```diff
@@ -388,39 +389,44 @@ private void createTransformers2(Network network, MatpowerModel model, Context c
             Bus bus1 = t1.getBusView().getBus();
             Bus bus2 = t2.getBusView().getBus();
             if (isExported(bus1) && isExported(bus2)) {
-                VoltageLevel vl1 = t1.getVoltageLevel();
-                VoltageLevel vl2 = t2.getVoltageLevel();
-                MBranch mBranch = new MBranch();
-                mBranch.setFrom(context.mBusesNumbersByIds.get(bus1.getId()));
-                mBranch.setTo(context.mBusesNumbersByIds.get(bus2.getId()));
-                mBranch.setStatus(CONNECTED_STATUS);
-                double r = twt.getR();
-                double x = twt.getX();
-                double b = twt.getB();
-                double rho = (twt.getRatedU2() / vl2.getNominalV()) / (twt.getRatedU1() / vl1.getNominalV());
-                var rtc = twt.getRatioTapChanger();
-                if (rtc != null) {
-                    rho *= rtc.getCurrentStep().getRho();
-                    r *= 1 + rtc.getCurrentStep().getR() / 100;
-                    x *= 1 + rtc.getCurrentStep().getX() / 100;
-                    b *= 1 + rtc.getCurrentStep().getB() / 100;
-                }
-                var ptc = twt.getPhaseTapChanger();
-                if (ptc != null) {
-                    mBranch.setPhaseShiftAngle(-ptc.getCurrentStep().getAlpha());
-                    rho *= ptc.getCurrentStep().getRho();
-                    r *= 1 + ptc.getCurrentStep().getR() / 100;
-                    x *= 1 + ptc.getCurrentStep().getX() / 100;
-                    b *= 1 + ptc.getCurrentStep().getB() / 100;
+                if (!bus1.getId().equals(bus2.getId())) {
```

Comment:
Do you really think that it can happen with transformers?

Reply:
I never had the case indeed